### ISSUE TITLE
False positive with `{**m}` on structural Mapping

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#3190]
Pyright on this file:

```python
from typing import List

class MyMapping:
    def keys(self) -> List[bool]:
        raise NotImplementedError

    def __getitem__(self, key: bool) -> float:
        raise NotImplementedError

def capybara(m: MyMapping):
    {**m}
```
produces `Expected mapping for dictionary unpack operator`.

But it works at runtime. Any class with `keys` and `__getitem__` is accepted.

I encountered this in real-world code that used `pandas.Series`.

**Comment by erictraut** [comment]
Are those the only two methods required for the dictionary unpack operator to work?

I don't see a standard protocol definition that defines these requirements. Am I missing it, or is this an omission in the typeshed stubs?

**Comment by JelleZijlstra** [comment]
Yes, only those two. I don't think this is documented; I got it from reading the code at https://github.com/python/cpython/blob/03c2a36b2bd2d4469160d1607619ee144175d753/Objects/dictobject.c#L2811.

The protocol is `_typeshed.SupportsKeysAndGetItem`. It's not in typing or collections.abc.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.230, which I just published. It will also be included in the next release of pylance.
